### DISCUSSION TITLE
How is the .angle property of grainBoundary/orientation/rotation class implemented?

### DISCUSSION THREAD

**Discussion by Artaher** [mtex-toolbox/mtex#2190] [Ask Anything]
Hi MTEX team,

I'm comparing **.angle** property of grainBoundary/orientation/rotation class with disorientation algorithm $O_Bg_B (O_Ag_A)^{-1}$ based on rotation matrix mentioned in [Misorientation - Wikipedia](https://en.wikipedia.org/wiki/Misorientation). I noticed that the default case in [mtex/geometry/@orientation/angle.m at develop · mtex-toolbox/mtex](https://github.com/mtex-toolbox/mtex/blob/develop/geometry/%40orientation/angle.m) , i.e. the last part of "if" statement, takes into account the proper symmetry transformations of the Laue group. The returned omega is calculated based on that the real part of the unit quaternion is equal to $\cos(\frac{\theta}{2})$. 

By reading grainBoundary.m, orientation.m, rotation.m and dynOption.m, I guess that the implementation of gB.misorientation.angle in [Misorientations at grain boundaries | MTEX](https://mtex-toolbox.github.io/BoundaryMisorientations.html) is assigned by the struct stored in the dynamic property **opt**. But I did not find a statement like `xxx.opt = angle()` in the source codes of repository.

After testing, it was found that for the same EBSD data, the distribution given by MTEX's .angle was significantly different from that given by the disorientation algorithm.

**Comment by ralfHielscher** [comment]
Hi,

I was not really able to understand your confusion. Here some answers that might help:
1. `orientation/angle` indeed considers only the rotational part of the symmetry groups. This might give incorrect results in cases of symmetry groups like mm2  that are neither Laue groups nor purely rotational groups. In practice I'm not aware of anybody using with such symmetries.
2. the misorientation angles are not stored in `xxx.opt.angle` but rather computed on demand.
3. If you feel that Matlab gives you different results than any other implementation - it would be very helpful if you could provide more information.

All the best,
Ralf.
